### PR TITLE
Add support for sending mails through SES

### PR DESCRIPTION
**Pull request by kulpreet**
Assumes that the Mailer#send_mail will be called from a background
worker task

### CODE REVIEW

File: src/mailer.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const nodemailer = require('nodemailer');
+const { isEmpty } = require('lodash')
+
+class Mailer {
+
+  constructor(sender, reply_to) {
+    if (isEmpty(process.env.SMTP_USER)) {
+      throw('Missing smtp user in environment variable')
+    }
+    if (isEmpty(process.env.SMTP_USER)) {
```
Comment (fazo96) [inline src/mailer.js:10]: typo here

File: src/mailer.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    // setup email data with unicode symbols
+    let mailOptions = {
+      from: this.sender,
+      to: to,
+      subject: subject,
+      text: text_body,
+      html: html_body
+    }
+    console.log(mailOptions)
```
Comment (fazo96) [inline src/mailer.js:43]: I think we should pass a logger for these so chatter can be turned off in tests, like we do in other projects

**Review by fazo96 — CHANGES_REQUESTED** [review]
All good except I think it'd be best to pass the smtp parameters to the constructor instead of reading them from the ENV directly. The reading from the ENV will happen at a higher level (CLI arg parsing for example)

We can also keep it like it is now, I don't think the use of ENV variables like this will be a problem, it just does not feel very clean

What do you think @kulpreet?

**Comment by kulpreet** [comment]
Agree totally with what you are saying. But the ChluIpfs constructor is so embedded in, it is difficult to disentangle options parsing for other modules, that is why I just hacked it as process.env for now :)

**Review by fazo96 — APPROVED** [review]

**Pull request merged**